### ISSUE TITLE
Client ignores host option, always connects to ::1

### ISSUE THREAD

**Issue by sammy007** [liftbridge-io/go-liftbridge#33]
```
addrs := []string{"some.server:9292", "some.another.server:9293"}
client, err := lift.Connect(addrs)
```
```
Error while dialing dial tcp [::1]:9293
```

Looks like client ignores host setting.

**Comment by tylertreat** [comment]
Hmm, that doesn't sound right as it should attempt to dial the address provided (https://github.com/liftbridge-io/go-liftbridge/blob/master/client.go#L235). Will take a look...

**Comment by sammy007** [comment]
I incorrectly reported the issue, but it's still there:

```go
package main

import (
	"fmt"

	"context"
	lift "github.com/liftbridge-io/go-liftbridge"
	proto "github.com/liftbridge-io/liftbridge-api/go"
)

func main() {
	addrs := []string{"some.node:9292"}
	client, err := lift.Connect(addrs)
	if err != nil {
		panic(err)
	}
	defer client.Close()

	ctx := context.Background()
	if err := client.Subscribe(ctx, "foo-stream", func(msg *proto.Message, err error) {
		if err != nil {
			panic(err)
		}
		fmt.Println(msg.Offset, string(msg.Value))
	}, lift.StartAtEarliestReceived()); err != nil {
		panic(err) # !!! PANICS HERE.
	}

	<-ctx.Done()
}
```

```
goroutine 1 [running]:
main.main()
	/Users/me/code/toys/lb.go:27
```

It panics after `Subscribe`. Of course if wrong port/host specified it panics right after connect.

**Comment by sammy007** [comment]
Well, the issue is that if I set `listen: 0.0.0.0:9292` in config and connect to it from outside client will try to use `0.0.0.0` and panics because obviously I have nothing listening on that port on client machine. If I set external instead of `0.0.0.0` client can perform `Subscribe`. So when setting up cluster need to keep that in mind and explicitly specify external IPs and never use that `0.0.0.0` trick or localhost for some nodes. Not sure that is right behaviour though.
